### PR TITLE
refactor/chore: Move `Capabilities` into own module + Improve readability of `leaf_node.rs`.

### PR DESCRIPTION
**Pull request by duesee**
This is a pure refactoring. Thus, I used `pub use capabilities::*` to re-export and `pub(crate)` for access control. Edit: Also added two `TODO`s. Edit: While at it, I also improved the readability of the `leaf_node.rs` file. I'll split the file into distinct typestate modules in a follow-up. (This PR is best reviewed commit-by-commit.)

### CODE REVIEW

File: openmls/src/treesync/node/leaf_node.rs [hunk 1]
```diff
@@ -797,13 +619,17 @@ impl LeafNode {
             .capabilities
             .extensions
             .contains(extension_type)
-            || default_extensions().iter().any(|et| et == extension_type)
+            || capabilities::default_extensions()
```
Comment (franziskuskiefer) [inline openmls/src/treesync/node/leaf_node.rs:441]: There's a `pub use capabilities::*` so why this change?

File: openmls/src/treesync/node/leaf_node.rs [hunk 2]
```diff
@@ -690,49 +503,103 @@ impl LeafNode {
...
-impl<'a> Verifiable for VerifiableLeafNodeTbs<'a> {
-    fn unsigned_payload(&self) -> Result<Vec<u8>, tls_codec::Error> {
-        self.tbs.tls_serialize_detached()
-    }
+pub type ParentHash = VLBytes;
 
-    fn signature(&self) -> &Signature {
-        self.signature
-    }
+// -------------------------------------------------------------------------------------------------
```
Comment (franziskuskiefer) [inline openmls/src/treesync/node/leaf_node.rs:568]: I'm not sure I understand what these are supposed to be.
If there's that much code here that something like a divider is needed, it should go somewhere else and not add even more content (even if it's a comment).

File: openmls/src/group/core_group/staged_commit.rs [hunk 3]
```diff
@@ -5,7 +5,7 @@ use crate::framing::mls_content::FramedContentBody;
 use crate::treesync::errors::TreeSyncAddLeaf;
 use crate::treesync::node::encryption_keys::EncryptionKeyPair;
 use crate::treesync::node::leaf_node::{
-    LeafNodeTbs, OpenMlsLeafNode, TreeInfoTbs, VerifiableLeafNodeTbs,
+    LeafNodeTbs, OpenMlsLeafNode, TreeInfo, VerifiableLeafNodeTbs,
```
Comment (franziskuskiefer) [inline openmls/src/group/core_group/staged_commit.rs]: This is an unnecessary change and unrelated to moving code around.

**Review by franziskuskiefer — CHANGES_REQUESTED** [review]
Some unnecessary changes in here. Otherwise lgtm.

File: openmls/src/group/core_group/staged_commit.rs (hunk 3 above)
Comment (duesee) [reply openmls/src/group/core_group/staged_commit.rs]: Can we keep this change? The `TreeInfo` is not signed and this PR is about cleaning up.

File: openmls/src/treesync/node/leaf_node.rs (hunk 2 above)
Comment (duesee) [reply openmls/src/treesync/node/leaf_node.rs:568]: It separates `LeafNode`, `LeafNodeTbs`, etc. code. Not sure if we want own modules for this. I can delete them.

File: openmls/src/group/core_group/staged_commit.rs (hunk 3 above)
Comment (duesee) [reply openmls/src/group/core_group/staged_commit.rs]: We discussed this offline and I'll revert this commit for now.

**Review by franziskuskiefer — APPROVED** [review]
